web/client/components/TOC/fragments/settings/FeatureInfoEditor.jsx

```diff
@@ -31,38 +32,74 @@ Quill.register({
  * @prop {bool} enableIFrameModule enable iframe in editor, default true
  */
 
-module.exports = ({onShowEditor = () => {}, showEditor, element = {}, onChange = () => {}, enableIFrameModule = true}) =>(
-    <Portal>
-        <ResizableModal
-            fade
-            show={showEditor}
-            title={<Message msgId="layerProperties.editCustomFormat"/>}
-            size="lg"
-            showFullscreen
-            clickOutEnabled={false}
-            onClose={() => onShowEditor(!showEditor)}
-            buttons={[
-                {
-                    bsStyle: 'primary',
-                    text: <Message msgId="close"/>,
-                    onClick: () => onShowEditor(!showEditor)
-                }
-            ]}>
-            <div id="ms-template-editor" className="ms-editor">
-                <ReactQuill
-                    bounds="#ms-template-editor"
-                    modules={enableIFrameModule ? {
-                        resizeModule: {},
-                        toolbar: toolbarConfig
-                    } : {}}
-                    defaultValue={element.featureInfo && element.featureInfo.template || ' '}
-                    onChange={template => {
-                        onChange('featureInfo', {
-                            ...(element && element.featureInfo || {}),
-                            template
-                        });
-                    }}/>
-            </div>
-        </ResizableModal>
-    </Portal>
-);
+class FeatureInfoEditor extends React.Component {
+
+    static propTypes = {
+        showEditor: PropTypes.bool,
+        element: PropTypes.object,
+        onChange: PropTypes.func,
+        onShowEditor: PropTypes.func,
+        enableIFrameModule: PropTypes.bool
+    };
+
+    static defaultProps = {
+        showEditor: false,
+        element: {},
+        enableIFrameModule: false,
+        onChange: () => {},
+        onShowEditor: () => {}
+    };
+
+    state = { };
```

Comment:
I would put an empty (null?) template property in the initial state. It's more clear what the state contains, in my opinion.